x-pack/plugins/fleet/server/routes/app/index.ts

```diff
@@ -22,22 +27,11 @@ export const getCheckPermissionsHandler: RequestHandler = async (context, reques
   if (!appContextService.getSecurityLicense().isEnabled()) {
     return response.ok({ body: missingSecurityBody });
   } else {
-    const security = appContextService.getSecurity();
-    const user = security.authc.getCurrentUser(request);
-
-    // Defensively handle situation where user is undefined (should only happen when ES security is disabled)
-    // This should be covered by the `getSecurityLicense().isEnabled()` check above, but we leave this for robustness.
-    if (!user) {
-      return response.ok({
-        body: missingSecurityBody,
-      });
-    }
-
-    if (!user?.roles.includes('superuser')) {
```

Comment:
@paul-tavares I've confirmed that this breaks the requirement to continue requiring superuser for the Endpoint UI. I think the best course of action here would be to implement your own dedicated endpoint for checking superuser and only using this endpoint to verify that security features are enabled.

Reply:
@joshdover ,
This seems timely 😄 
I just [merged in a change](https://github.com/elastic/kibana/pull/123425) this morning that adds the check on our side for superuser. can you pull `upstream/main`  into your branch and see if you still get the failures?

Reply:
@joshdover I can check it out if needed - just let me know.
Looks like there is a build going now, so once it finishes I can check. We have tests that pull in fleet data as well as actually access Fleet pages (package policy edit flow) to ensure our content is displayed. Those tests are here:

https://github.com/elastic/kibana/tree/e528e413609ce4a8da19f144689d7c680ab52031/x-pack/test/security_solution_endpoint/apps/endpoint

Reply:
@paul-tavares Would you mind testing this? With a user configured without superuser role, but with Fleet privileges, I'm able to create an endpoint policy via the Security app on this PR and get the Agent enroll flow UI to display as well. I didn't test further, but I'm not sure where it's supposed to block me from using the Security app here without superuser.

Reply:
@paul-tavares @joshdover I'll pull this down to check on it.  My guess is that our UI is still being displayed based on Fleet being enabled, when in reality, we need to disable it based on `superuser`.  Paul, was that a part of your `authz` changes?

Reply:
me culpa!

I got a PR up to fix this on our side. We were still checking for `fleet.all`, thus why its probably still allowing you to access endpoint management. PR is here: https://github.com/elastic/kibana/pull/124028